config/thisroot.sh

```diff
@@ -162,8 +162,12 @@ if [ -n "${ROOTSYS}" ] ; then
    old_rootsys=${ROOTSYS}
 fi
 
+if [ "x${BASH_ARGV}" = "x" ]; then
+   SOURCE="$(readlink -f "$0")"
```

Comment:
IIUC, osx does not have readlink available by default.

The other concern I have is that there might be other places in that very script which assume not being a symlink.

Reply:
I don't know what osx has... I know that there is a bug/problem. If you are worried that it fails on MAC, encapsulate it in:

if [ "$(which readlink)" ]; then

I think a problem is better fixed on one operating system than on none. How to fix MAC stuff, MAC users have to contribute.

Well... I am not sure if there are other places. I have tested it and it worked for me. Again, better a fix that works sometimes than no fix (which works never). If then "other places" pop up with problems, they can be fixed.

Btw, talking about "osx does not have"... what if the shell used does not support BASH_ARGV?

Reply:
> I think a problem is better fixed on one operating system than on none.

That's a source of inconsistency - our users would rightfully expect that ROOT behaves the same on Linux and MacOS. So why not try to fix it on MacOS, too? https://stackoverflow.com/questions/1055671/how-can-i-get-the-behavior-of-gnus-readlink-f-on-a-mac might help - but a simple solution in the form of `while readlink ${ARG0} | read ARG0SYM; do...` could already do the trick?

Reply:
I am super happy if somebody fixes the problem also for MacOSX, but I do neither have experience with nor access to MacOSX. I will not start implementing code that I can not test.